engine/consensus/approvals/verifying_assignment_collector.go

```diff
@@ -360,9 +360,15 @@ func (ac *VerifyingAssignmentCollector) RequestMissingApprovals(observation cons
 				)
 			}
 
+			nonce, err := rand.Uint64()
+			if err != nil {
+				log.Error().Err(err).
+					Msg("nonce generation failed during request missing approvals")
```

Comment:
```suggestion
				return fmt.Errorf("nonce generation failed during request missing approvals: %w", err)
```

Reply:
Good catch! 

Shouldn't this error also be returned, or do you think it's intended not to? https://github.com/onflow/flow-go/blob/tarak/randomness-part-6-consenus-engine/engine/consensus/approvals/verifying_assignment_collector.go#L377-L381

Reply:
For that linked case, logging and continuing is the best we can do, because `network.Conduit` doesn't differentiate error types.